<launch>

  <arg name="robot_model"                       default=""/>
  <arg name="robot_name"                        default="$(arg robot_model)"/>
  <arg name="base_link_frame"                   default="base_link"/>
  <arg name="use_world_frame"                   default="true"/>
  <arg name="external_urdf_loc"                 default=""/>
  <arg name="use_rviz"                          default="true"/>
  <arg name="mode_configs"                      default="$(find interbotix_xsturret_simple_interface)/config/modes.yaml"/>
  <arg name="threshold"                         default="0.75"/>
  <arg name="controller"                        default="ps4"/>
  <arg name="launch_driver"                     default="true"/>
  <arg name="use_sim"                           default="false"/>

  <include if="$(arg launch_driver)" file="$(find interbotix_xsturret_control)/launch/xsturret_control.launch">
    <arg name="robot_model"                       value="$(arg robot_model)"/>
    <arg name="robot_name"                        value="$(arg robot_name)"/>
    <arg name="base_link_frame"                   value="$(arg base_link_frame)"/>
    <arg name="use_world_frame"                   value="$(arg use_world_frame)"/>
    <arg name="external_urdf_loc"                 value="$(arg external_urdf_loc)"/>
    <arg name="use_rviz"                          value="$(arg use_rviz)"/>
    <arg name="mode_configs"                      value="$(arg mode_configs)"/>
    <arg name="use_sim"                           value="$(arg use_sim)"/>
  </include>

  <node
    pkg="joy"
    type="joy_node"
    name="joy_node"
    output="screen"
    ns="$(arg robot_name)" >
    <param name="dev"                             value="/dev/input/js0"/>
    <remap from="joy"                             to="commands/joy_raw"/>
  </node>

  <node
    pkg="interbotix_xsturret_simple_interface"
    type="xsturret_simple_interface_joy"
    name="xsturret_simple_interface_joy"
    output="screen"
    ns="$(arg robot_name)">
    <param name="threshold"                       value="$(arg threshold)"/>
    <param name="controller"                      value="$(arg controller)"/>
  </node>

  <node
    pkg="interbotix_xsturret_simple_interface"
    type="xsturret_simple_interface_gui"
    name="xsturret_simple_interface_gui"
    output="screen"
    ns="$(arg robot_name)">
  </node>

</launch>
